### DISCUSSION TITLE
manage.py get_email failing with production settings

### DISCUSSION THREAD

**Discussion by trainor1** [django-helpdesk/django-helpdesk#1110] [Q&A]
Hi All,

I have deployed to my production server.  I am not having success getting _manage.py get_email_ to run there.  I will include the settings differences and messages below.  I did realize that I never tested _manage.py get_email_ on my development machine using the production settings.  When I did, I discovered the same error.  So, I don't think that this is a server configuration error.

The major difference between the two settings files is that the development settings use sqlite3 and the production settings use Postgres.  I have tested m_anage.py runserver_ with both development and production settings.  The Web app works fine.  So, I can access Postgress OK when I am running the development server.  But, something is going wrong when I try to do _get_emai_l while connected to Postgres.

**Here is a copy of the successful output when running get_email with development settings:**

manage.py@django_apps_02 > get_email
bash -cl "/Users/trainor1/opt/anaconda3/envs/e4_django_helpdesk/bin/python /Users/trainor1/Library/Application\ Support/JetBrains/Toolbox/apps/PyCharm-P/ch-0/232.8660.197/PyCharm.app/Contents/plugins/python/helpers/pycharm/django_manage.py get_email /Users/trainor1/____my_python_repos/django_apps_02"
Tracking file by folder pattern:  migrations
  50:49.28 > b'KJPO1 AUTHENTICATE XOAUTH2'
  50:49.31 < b'+ '
  50:49.31 write literal size 2152
  50:50.61 < b'KJPO1 OK AUTHENTICATE completed.'
  50:50.61 > b'KJPO2 SELECT INBOX'
  50:50.88 < b'* 0 EXISTS'
  50:50.88 < b'* 0 RECENT'
  50:50.89 < b'* FLAGS (\\Seen \\Answered \\Flagged \\Deleted \\Draft $MDNSent)'
  50:50.89 < b'* OK [PERMANENTFLAGS (\\Seen \\Answered \\Flagged \\Deleted \\Draft $MDNSent)] Permanent flags'
  50:50.89 < b'* OK [UIDVALIDITY 14] UIDVALIDITY value'
  50:50.89 < b'* OK [UIDNEXT 50] The next unique identifier value'
  50:50.89 < b'KJPO2 OK [READ-WRITE] SELECT completed.'
  50:50.89 > b'KJPO3 SEARCH NOT DELETED'
  50:50.96 < b'* SEARCH'
  50:50.96 < b'KJPO3 OK SEARCH completed.'
  50:50.96 > b'KJPO4 EXPUNGE'
  50:51.03 < b'* 0 EXISTS'
  50:51.13 < b'KJPO4 OK EXPUNGE completed.'
  50:51.13 > b'KJPO5 CLOSE'
  50:51.21 < b'KJPO5 OK CLOSE completed.'
  50:51.21 > b'KJPO6 LOGOUT'
  50:51.28 < b'* BYE Microsoft Exchange Server IMAP4 server signing off.'
  50:51.28 BYE response: b'Microsoft Exchange Server IMAP4 server signing off.'
  50:51.63 > b'EKFD1 AUTHENTICATE XOAUTH2'
  50:51.66 < b'+ '
  50:51.66 write literal size 2148
  50:53.37 < b'EKFD1 OK AUTHENTICATE completed.'
  50:53.37 > b'EKFD2 SELECT INBOX'
  50:53.70 < b'* 0 EXISTS'
  50:53.70 < b'* 0 RECENT'
  50:53.70 < b'* FLAGS (\\Seen \\Answered \\Flagged \\Deleted \\Draft $MDNSent)'
  50:53.70 < b'* OK [PERMANENTFLAGS (\\Seen \\Answered \\Flagged \\Deleted \\Draft $MDNSent)] Permanent flags'
  50:53.70 < b'* OK [UIDVALIDITY 14] UIDVALIDITY value'
  50:53.70 < b'* OK [UIDNEXT 148] The next unique identifier value'
  50:53.70 < b'EKFD2 OK [READ-WRITE] SELECT completed.'
  50:53.70 > b'EKFD3 SEARCH NOT DELETED'
  50:53.78 < b'* SEARCH'
  50:53.78 < b'EKFD3 OK SEARCH completed.'
  50:53.78 > b'EKFD4 EXPUNGE'
  50:53.86 < b'* 0 EXISTS'
  50:53.96 < b'EKFD4 OK EXPUNGE completed.'
  50:53.96 > b'EKFD5 CLOSE'
  50:54.06 < b'EKFD5 OK CLOSE completed.'
  50:54.06 > b'EKFD6 LOGOUT'
  50:54.14 < b'* BYE Microsoft Exchange Server IMAP4 server signing off.'
  50:54.14 BYE response: b'Microsoft Exchange Server IMAP4 server signing off.'

**And here is a copy of the unsuccessful output when I run get_email with production settings:**

manage.py@django_apps_02 > get_email
bash -cl "/Users/trainor1/opt/anaconda3/envs/e4_django_helpdesk/bin/python /Users/trainor1/Library/Application\ Support/JetBrains/Toolbox/apps/PyCharm-P/ch-0/232.8660.197/PyCharm.app/Contents/plugins/python/helpers/pycharm/django_manage.py get_email /Users/trainor1/____my_python_repos/django_apps_02"
Tracking file by folder pattern:  migrations
  52:38.15 > b'MOJP1 AUTHENTICATE XOAUTH2'
  52:38.18 < b'+ '
  52:38.18 write literal size 2152
  52:39.34 < b'MOJP1 OK AUTHENTICATE completed.'
  52:39.34 > b'MOJP2 SELECT INBOX'
  52:39.44 < b'* 0 EXISTS'
  52:39.44 < b'* 0 RECENT'
  52:39.44 < b'* FLAGS (\\Seen \\Answered \\Flagged \\Deleted \\Draft $MDNSent)'
  52:39.44 < b'* OK [PERMANENTFLAGS (\\Seen \\Answered \\Flagged \\Deleted \\Draft $MDNSent)] Permanent flags'
  52:39.44 < b'* OK [UIDVALIDITY 14] UIDVALIDITY value'
  52:39.44 < b'* OK [UIDNEXT 50] The next unique identifier value'
  52:39.44 < b'MOJP2 OK [READ-WRITE] SELECT completed.'
  52:39.44 > b'MOJP3 SEARCH NOT DELETED'
  52:39.51 < b'* SEARCH'
  52:39.51 < b'MOJP3 OK SEARCH completed.'
  52:39.51 > b'MOJP4 EXPUNGE'
  52:39.59 < b'* 0 EXISTS'
  52:39.68 < b'MOJP4 OK EXPUNGE completed.'
  52:39.68 > b'MOJP5 CLOSE'
  52:39.76 < b'MOJP5 OK CLOSE completed.'
  52:39.76 > b'MOJP6 LOGOUT'
  52:39.83 < b'* BYE Microsoft Exchange Server IMAP4 server signing off.'
  52:39.83 BYE response: b'Microsoft Exchange Server IMAP4 server signing off.'
  52:40.25 > b'NOFN1 AUTHENTICATE XOAUTH2'
  52:40.27 < b'+ '
  52:40.27 write literal size 2152
  52:41.23 < b'NOFN1 OK AUTHENTICATE completed.'
  52:41.23 > b'NOFN2 SELECT INBOX'
  52:41.30 < b'* 0 EXISTS'
  52:41.30 < b'* 0 RECENT'
  52:41.30 < b'* FLAGS (\\Seen \\Answered \\Flagged \\Deleted \\Draft $MDNSent)'
  52:41.30 < b'* OK [PERMANENTFLAGS (\\Seen \\Answered \\Flagged \\Deleted \\Draft $MDNSent)] Permanent flags'
  52:41.30 < b'* OK [UIDVALIDITY 14] UIDVALIDITY value'
  52:41.30 < b'* OK [UIDNEXT 23] The next unique identifier value'
  52:41.30 < b'NOFN2 OK [READ-WRITE] SELECT completed.'
  52:41.30 > b'NOFN3 SEARCH NOT DELETED'
  52:41.35 < b'* SEARCH'
  52:41.35 < b'NOFN3 OK SEARCH completed.'
  52:41.35 > b'NOFN4 EXPUNGE'
  52:41.40 < b'* 0 EXISTS'
  52:41.47 < b'NOFN4 OK EXPUNGE completed.'
  52:41.47 > b'NOFN5 CLOSE'
  52:41.52 < b'NOFN5 OK CLOSE completed.'
  52:41.52 > b'NOFN6 LOGOUT'
  52:41.57 < b'* BYE Microsoft Exchange Server IMAP4 server signing off.'
  52:41.57 BYE response: b'Microsoft Exchange Server IMAP4 server signing off.'
  52:41.95 > b'BPBI1 AUTHENTICATE XOAUTH2'
  52:41.97 < b'+ '
  52:41.97 write literal size 2140
  52:42.86 < b'BPBI1 OK AUTHENTICATE completed.'
  52:42.86 > b'BPBI2 SELECT'
  52:42.93 < b'BPBI2 BAD Command Argument Error. 12'
  52:42.93 BAD response: b'Command Argument Error. 12'
Traceback (most recent call last):
  File "/Users/trainor1/Library/Application Support/JetBrains/Toolbox/apps/PyCharm-P/ch-0/232.8660.197/PyCharm.app/Contents/plugins/python/helpers/pycharm/django_manage.py", line 52, in <module>
    run_command()
  File "/Users/trainor1/Library/Application Support/JetBrains/Toolbox/apps/PyCharm-P/ch-0/232.8660.197/PyCharm.app/Contents/plugins/python/helpers/pycharm/django_manage.py", line 46, in run_command
    run_module(manage_file, None, '__main__', True)
  File "<frozen runpy>", line 226, in run_module
  File "<frozen runpy>", line 98, in _run_module_code
  File "<frozen runpy>", line 88, in _run_code
  File "/Users/trainor1/____my_python_repos/django_apps_02/manage.py", line 22, in <module>
    main()
  File "/Users/trainor1/____my_python_repos/django_apps_02/manage.py", line 18, in main
    execute_from_command_line(sys.argv)
  File "/Users/trainor1/opt/anaconda3/envs/e4_django_helpdesk/lib/python3.11/site-packages/django/core/management/__init__.py", line 442, in execute_from_command_line
    utility.execute()
  File "/Users/trainor1/opt/anaconda3/envs/e4_django_helpdesk/lib/python3.11/site-packages/django/core/management/__init__.py", line 436, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/Users/trainor1/opt/anaconda3/envs/e4_django_helpdesk/lib/python3.11/site-packages/django/core/management/base.py", line 412, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/Users/trainor1/opt/anaconda3/envs/e4_django_helpdesk/lib/python3.11/site-packages/django/core/management/base.py", line 458, in execute
    output = self.handle(*args, **options)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/trainor1/opt/anaconda3/envs/e4_django_helpdesk/lib/python3.11/site-packages/helpdesk/management/commands/get_email.py", line 36, in handle
    process_email(quiet=quiet)
  File "/Users/trainor1/opt/anaconda3/envs/e4_django_helpdesk/lib/python3.11/site-packages/helpdesk/email.py", line 93, in process_email
    process_queue(q, logger=logger)
  File "/Users/trainor1/opt/anaconda3/envs/e4_django_helpdesk/lib/python3.11/site-packages/helpdesk/email.py", line 394, in process_queue
    mail_defaults[email_box_type]['sync'](q, logger, server)
  File "/Users/trainor1/opt/anaconda3/envs/e4_django_helpdesk/lib/python3.11/site-packages/helpdesk/email.py", line 263, in imap_oauth_sync
    server.select(q.email_box_imap_folder)
  File "/Users/trainor1/opt/anaconda3/envs/e4_django_helpdesk/lib/python3.11/imaplib.py", line 756, in select
    typ, dat = self._simple_command(name, mailbox)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/trainor1/opt/anaconda3/envs/e4_django_helpdesk/lib/python3.11/imaplib.py", line 1230, in _simple_command
    return self._command_complete(name, self._command(name, *args))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/trainor1/opt/anaconda3/envs/e4_django_helpdesk/lib/python3.11/imaplib.py", line 1055, in _command_complete
    raise self.error('%s command error: %s %s' % (name, typ, data))
imaplib.IMAP4.error: SELECT command error: BAD [b'Command Argument Error. 12']

**-------**

Finally, I am attaching copies of my settings files with keys obscured below.

I would appreciate any ideas about what might be causing this problem when using Postgres and production settings.

Thanks for your help.

Kevin

[base.py.txt](https://github.com/django-helpdesk/django-helpdesk/files/12263827/base.py.txt)
[development.py.txt](https://github.com/django-helpdesk/django-helpdesk/files/12263828/development.py.txt)
[production.py.txt](https://github.com/django-helpdesk/django-helpdesk/files/12263829/production.py.txt)

**Comment by brucegibbins** [comment]
Hi

How are you setting up which settings.py file to use?

**Comment by uhurusurfa** [comment]
Have you set up your queues properly?
Looks like it is this call from email.py that is failing because the argument value is invalid:
```
        # Select the Inbound Mailbox folder
        server.select(q.email_box_imap_folder)
```

**Reply by trainor1** [reply]
In the example that I sent to you, I was choosing the settings.py file using the PyCharm wizard.  But, I get the same results when I run this in the terminal and choose the setting.py file via --select...

**Reply by trainor1** [reply]
The databases are different.  I created the production database from scratch.  I did not copy it.  I guess that it is possible that I did not set up the queues properly.  I will check that now.

**Reply by trainor1** [reply]
You are right about this.  I turned off all queues but one and the problem resolved.  Now, I just have to find the queue with ad inbound email config values.  Thanks for pointing me in this direction.

**Reply by brucegibbins** [reply]
The IMAP SELECT command is typically used to select then inbox folder that is to be read by the subsequent IMAP commands. You may be using an non-existent folder.

**Comment by trainor1** [comment]
Hi Bruce,

In the example that I posted, I was picking the settings file using PyCharm configuration features.  Here, I am doing it in the terminal using --settings:

**HERE IS THE DEVELOPMENT VERSION:**

(e4_django_helpdesk) trainor1@lism01-trainor1 django_apps_02 % python manage.py get_email --settings=django_apps_02.settings.development
  47:33.48 > b'CFMB1 AUTHENTICATE XOAUTH2'
  47:33.50 < b'+ '
  47:33.50 write literal size 2152
  47:34.55 < b'CFMB1 OK AUTHENTICATE completed.'
  47:34.55 > b'CFMB2 SELECT INBOX'
  47:34.78 < b'* 0 EXISTS'
  47:34.78 < b'* 0 RECENT'
  47:34.78 < b'* FLAGS (\\Seen \\Answered \\Flagged \\Deleted \\Draft $MDNSent)'
  47:34.78 < b'* OK [PERMANENTFLAGS (\\Seen \\Answered \\Flagged \\Deleted \\Draft $MDNSent)] Permanent flags'
  47:34.78 < b'* OK [UIDVALIDITY 14] UIDVALIDITY value'
  47:34.78 < b'* OK [UIDNEXT 50] The next unique identifier value'
  47:34.78 < b'CFMB2 OK [READ-WRITE] SELECT completed.'
  47:34.78 > b'CFMB3 SEARCH NOT DELETED'
  47:34.86 < b'* SEARCH'
  47:34.86 < b'CFMB3 OK SEARCH completed.'
  47:34.86 > b'CFMB4 EXPUNGE'
  47:34.93 < b'* 0 EXISTS'
  47:35.02 < b'CFMB4 OK EXPUNGE completed.'
  47:35.02 > b'CFMB5 CLOSE'
  47:35.10 < b'CFMB5 OK CLOSE completed.'
  47:35.10 > b'CFMB6 LOGOUT'
  47:35.18 < b'* BYE Microsoft Exchange Server IMAP4 server signing off.'
  47:35.18 BYE response: b'Microsoft Exchange Server IMAP4 server signing off.'
  47:35.64 > b'OCNM1 AUTHENTICATE XOAUTH2'
  47:35.67 < b'+ '
  47:35.67 write literal size 2148
  47:36.89 < b'OCNM1 OK AUTHENTICATE completed.'
  47:36.89 > b'OCNM2 SELECT INBOX'
  47:37.15 < b'* 0 EXISTS'
  47:37.15 < b'* 0 RECENT'
  47:37.15 < b'* FLAGS (\\Seen \\Answered \\Flagged \\Deleted \\Draft $MDNSent)'
  47:37.15 < b'* OK [PERMANENTFLAGS (\\Seen \\Answered \\Flagged \\Deleted \\Draft $MDNSent)] Permanent flags'
  47:37.15 < b'* OK [UIDVALIDITY 14] UIDVALIDITY value'
  47:37.15 < b'* OK [UIDNEXT 149] The next unique identifier value'
  47:37.15 < b'OCNM2 OK [READ-WRITE] SELECT completed.'
  47:37.15 > b'OCNM3 SEARCH NOT DELETED'
  47:37.23 < b'* SEARCH'
  47:37.23 < b'OCNM3 OK SEARCH completed.'
  47:37.23 > b'OCNM4 EXPUNGE'
  47:37.31 < b'* 0 EXISTS'
  47:37.40 < b'OCNM4 OK EXPUNGE completed.'
  47:37.40 > b'OCNM5 CLOSE'
  47:37.48 < b'OCNM5 OK CLOSE completed.'
  47:37.48 > b'OCNM6 LOGOUT'
  47:37.56 < b'* BYE Microsoft Exchange Server IMAP4 server signing off.'
  47:37.56 BYE response: b'Microsoft Exchange Server IMAP4 server signing off.'
(e4_django_helpdesk) trainor1@lism01-trainor1 django_apps_02 % 

**HERE IS THE PRODUCTION VERSION:**

(e4_django_helpdesk) trainor1@lism01-trainor1 django_apps_02 % python manage.py get_email --settings=django_apps_02.settings.production
  51:44.36 > b'IMKA1 AUTHENTICATE XOAUTH2'
  51:44.39 < b'+ '
  51:44.39 write literal size 2152
  51:45.55 < b'IMKA1 OK AUTHENTICATE completed.'
  51:45.55 > b'IMKA2 SELECT INBOX'
  51:45.66 < b'* 0 EXISTS'
  51:45.66 < b'* 0 RECENT'
  51:45.66 < b'* FLAGS (\\Seen \\Answered \\Flagged \\Deleted \\Draft $MDNSent)'
  51:45.66 < b'* OK [PERMANENTFLAGS (\\Seen \\Answered \\Flagged \\Deleted \\Draft $MDNSent)] Permanent flags'
  51:45.66 < b'* OK [UIDVALIDITY 14] UIDVALIDITY value'
  51:45.66 < b'* OK [UIDNEXT 50] The next unique identifier value'
  51:45.66 < b'IMKA2 OK [READ-WRITE] SELECT completed.'
  51:45.66 > b'IMKA3 SEARCH NOT DELETED'
  51:45.74 < b'* SEARCH'
  51:45.74 < b'IMKA3 OK SEARCH completed.'
  51:45.74 > b'IMKA4 EXPUNGE'
  51:45.82 < b'* 0 EXISTS'
  51:45.93 < b'IMKA4 OK EXPUNGE completed.'
  51:45.93 > b'IMKA5 CLOSE'
  51:46.01 < b'IMKA5 OK CLOSE completed.'
  51:46.01 > b'IMKA6 LOGOUT'
  51:46.09 < b'* BYE Microsoft Exchange Server IMAP4 server signing off.'
  51:46.09 BYE response: b'Microsoft Exchange Server IMAP4 server signing off.'
  51:46.53 > b'JBHD1 AUTHENTICATE XOAUTH2'
  51:46.56 < b'+ '
  51:46.56 write literal size 2152
  51:47.58 < b'JBHD1 OK AUTHENTICATE completed.'
  51:47.58 > b'JBHD2 SELECT INBOX'
  51:47.96 < b'* 0 EXISTS'
  51:47.96 < b'* 0 RECENT'
  51:47.96 < b'* FLAGS (\\Seen \\Answered \\Flagged \\Deleted \\Draft $MDNSent)'
  51:47.96 < b'* OK [PERMANENTFLAGS (\\Seen \\Answered \\Flagged \\Deleted \\Draft $MDNSent)] Permanent flags'
  51:47.96 < b'* OK [UIDVALIDITY 14] UIDVALIDITY value'
  51:47.96 < b'* OK [UIDNEXT 23] The next unique identifier value'
  51:47.96 < b'JBHD2 OK [READ-WRITE] SELECT completed.'
  51:47.96 > b'JBHD3 SEARCH NOT DELETED'
  51:48.02 < b'* SEARCH'
  51:48.02 < b'JBHD3 OK SEARCH completed.'
  51:48.02 > b'JBHD4 EXPUNGE'
  51:48.07 < b'* 0 EXISTS'
  51:48.15 < b'JBHD4 OK EXPUNGE completed.'
  51:48.15 > b'JBHD5 CLOSE'
  51:48.20 < b'JBHD5 OK CLOSE completed.'
  51:48.20 > b'JBHD6 LOGOUT'
  51:48.26 < b'* BYE Microsoft Exchange Server IMAP4 server signing off.'
  51:48.26 BYE response: b'Microsoft Exchange Server IMAP4 server signing off.'
  51:48.70 > b'OHOJ1 AUTHENTICATE XOAUTH2'
  51:48.73 < b'+ '
  51:48.73 write literal size 2140
  51:49.76 < b'OHOJ1 OK AUTHENTICATE completed.'
  51:49.76 > b'OHOJ2 SELECT'
  51:49.84 < b'OHOJ2 BAD Command Argument Error. 12'
  51:49.84 BAD response: b'Command Argument Error. 12'
Traceback (most recent call last):
  File "/Users/trainor1/____my_python_repos/django_apps_02/manage.py", line 22, in <module>
    main()
  File "/Users/trainor1/____my_python_repos/django_apps_02/manage.py", line 18, in main
    execute_from_command_line(sys.argv)
  File "/Users/trainor1/opt/anaconda3/envs/e4_django_helpdesk/lib/python3.11/site-packages/django/core/management/__init__.py", line 442, in execute_from_command_line
    utility.execute()
  File "/Users/trainor1/opt/anaconda3/envs/e4_django_helpdesk/lib/python3.11/site-packages/django/core/management/__init__.py", line 436, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/Users/trainor1/opt/anaconda3/envs/e4_django_helpdesk/lib/python3.11/site-packages/django/core/management/base.py", line 412, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/Users/trainor1/opt/anaconda3/envs/e4_django_helpdesk/lib/python3.11/site-packages/django/core/management/base.py", line 458, in execute
    output = self.handle(*args, **options)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/trainor1/opt/anaconda3/envs/e4_django_helpdesk/lib/python3.11/site-packages/helpdesk/management/commands/get_email.py", line 36, in handle
    process_email(quiet=quiet)
  File "/Users/trainor1/opt/anaconda3/envs/e4_django_helpdesk/lib/python3.11/site-packages/helpdesk/email.py", line 93, in process_email
    process_queue(q, logger=logger)
  File "/Users/trainor1/opt/anaconda3/envs/e4_django_helpdesk/lib/python3.11/site-packages/helpdesk/email.py", line 394, in process_queue
    mail_defaults[email_box_type]['sync'](q, logger, server)
  File "/Users/trainor1/opt/anaconda3/envs/e4_django_helpdesk/lib/python3.11/site-packages/helpdesk/email.py", line 263, in imap_oauth_sync
    server.select(q.email_box_imap_folder)
  File "/Users/trainor1/opt/anaconda3/envs/e4_django_helpdesk/lib/python3.11/imaplib.py", line 756, in select
    typ, dat = self._simple_command(name, mailbox)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/trainor1/opt/anaconda3/envs/e4_django_helpdesk/lib/python3.11/imaplib.py", line 1230, in _simple_command
    return self._command_complete(name, self._command(name, *args))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/trainor1/opt/anaconda3/envs/e4_django_helpdesk/lib/python3.11/imaplib.py", line 1055, in _command_complete
    raise self.error('%s command error: %s %s' % (name, typ, data))
imaplib.IMAP4.error: SELECT command error: BAD [b'Command Argument Error. 12']
(e4_django_helpdesk) trainor1@lism01-trainor1 django_apps_02 % 


Kevin

**Comment by trainor1** [comment]
Hi Bruce,

I am pleased to report that the error resolved when I turned off all but 1 queue to allow tickets via incoming mail.  There must be a configuration error in one of the other queues.  I turned on a second queue and that is reading email too.  So, I plan to turn on queues one by one until I find the offending queue.  Then, I will fix it.

On another front, emails are not being sent to the submitter when I update the ticket.  This is true despite me having checked the box on the comment form to make the update public.  I know that I had this working before.  Do I have to check the **Notify contacts when email updates arrive** checkbox on the queue config.  I don't remember doing that previously.  But, it is possible that I did and forgot.

Please suggest any ideas that would cause submitter to get a confirmation email back when they send in an email request BUT not get emails when I update the ticket.

Many thanks.

Kevin

**Reply by brucegibbins** [reply]
The queue idea was @uhurusurfa . I missed it looking at the email via mobile phone.

**Reply by brucegibbins** [reply]
There are config settings related to notifications. Suggest one of those needs tweaking.

**Comment by uhurusurfa** [comment]
Once the email parser changes have been merged we can added better checking and information logging on email inbox processing to make it easier to debug.

**Comment by trainor1** [comment]
I finally resolved the problem by identifying the offending queue.  I couldn't see the problem with the values on that queue for the life of me.  Given recent comments here, there was probably a problem with the mailbox name.  Perhaps instead of "INBOX" it said "INBOX ".  I deleted the queue and re-added it.  This led to excitement when the app did not want me to add a queue with the same name.  There were stray permissions left around from the original version.  I had to resort to deleting them with SQL.  That is an issue that I will post as a bug on another day.  I finally did get it to work.  I agree that improved error messages would help.

Thanks to everyone for your help.

**Comment by uhurusurfa** [comment]
#1128 will improve reliability of the email processing and supports a flag to use from the command line to more easily debug this sort of issue.
